### ISSUE TITLE
Bughunting

### ISSUE THREAD

**Issue by DrYool** [bgcval/bgcval2#1]
Just a note of a few minor issues I've noticed while working with the ALK suite. I've made a couple of changes to local (i.e. not committed) `key_files` as follows:

- Added `NorthAtlanticOcean`, `SouthAtlanticOcean`, `NorthPacificOcean`, `SouthPacificOcean` regions to request
- Altered `mld.yml` key_file to request: `model_vars      : mlotst somixhgt #;mlotstmax somxl010 somxzint1`

You can see the output from my setup's output here:

- https://gws-access.jasmin.ac.uk/public/esmeval/CompareReports/bgcval2/ayool/ukesm_bgcval_alkalinity/

Things I've noticed that are slightly glitchy are:

- Of the four regions mentioned above, only `SouthAtlanticOcean` seems to work, while the others just produce empty plots
- Although I tried to fix MLD, I've still got no output for it despite using variable names I got from the output files
- Some regions of some properties haven't been plotted at all, e.g. while other variables have them, salinity is missing some of the same regions requested

Otherwise, everything seems fine. It's looking good, actually, and BGCVal2 has been quite easy to modify. I'm sure these are minor glitches (... that I could possibly fix myself if I wasn't a Python hold-out).

**Comment by valeriupredoi** [comment]
@DrYool @ledm is currently still working in the local (and more up to date) repo https://github.com/valeriupredoi/bgcval2 - you mind if I transfer your issue there?

**Comment by ledm** [comment]
yeah, I've only just seen this issue, lol. Sorry. Is this still a problem or is it fixed now?

**Comment by DrYool** [comment]
Hi Lee,

I wasn't really sure how to fix it, so I've left it. I'm sure it's really simple to fix - a number of the masks are already there, so there must be some minor glitch with the ones that don't work.

In the context of the upcoming testing and spin-up of UKESM2, how easy is it to add another mask? To help tune the dust deposition, it'd be good to have a custom one that masked out regions we exclude for this calculation. But it might be a faff.

Cheers,

Andrew.


-----

Dr. Andrew Yool | MEDUSA lead developer
Room 256/19, National Oceanography Centre, European Way, Southampton, SO14 3ZH

t: +44 (0)23 8059 6339 | e: ***@***.***
noc.ac.uk<https://noc.ac.uk/> | Follow<https://linktr.ee/nocnews> | Support<https://noc.ac.uk/about-us/giving/donate> | Subscribe<https://noc.ac.uk/about-us/nocmail>

[https://noc.ac.uk/files/LOGO-2020---NOC-email-logo.png]<https://noc.ac.uk/>

My normal office hours are Tuesday-Friday | Monday is my non-working day
________________________________

**Comment by ledm** [comment]
It should be pretty easy to add new masks (provided they're squares or rectangles or combinations of squares and rectangles!)

Can you show us the changes you made to these files?


The masks are defined here, and kinda treats the model data like a csv file. Basically, model data is converted into columns for t, z, lat, lon and data, then you apply masks how you want. 
https://github.com/bgcval/bgcval2/blob/main/bgcval2/bgcvaltools/makeMask.py

**Comment by DrYool** [comment]
Hi Lee,

Thanks for getting back to me.

My recollection is that we just blocked out the Mediterranean Sea as that gets a disproportionate amount of dust deposition because of its proximity to the Sahara. I think that can be cut out with a single rectangular block, although I'll need to check. I'll report back once I have.

I'm not sure what you mean about change to the files. I don't think I've done anything there. Or do you mean to my BCGVal2 configuration / request files? I certainly added in a request for the other masks in those - but BGCVal2 just skipped over the masks it didn't know about.

Cheers,

Andrew.


-----

Dr. Andrew Yool | MEDUSA lead developer
Room 256/19, National Oceanography Centre, European Way, Southampton, SO14 3ZH

t: +44 (0)23 8059 6339 | e: ***@***.***
noc.ac.uk<https://noc.ac.uk/> | Follow<https://linktr.ee/nocnews> | Support<https://noc.ac.uk/about-us/giving/donate> | Subscribe<https://noc.ac.uk/about-us/nocmail>

[https://noc.ac.uk/files/LOGO-2020---NOC-email-logo.png]<https://noc.ac.uk/>

My normal office hours are Tuesday-Friday | Monday is my non-working day
________________________________

**Comment by ledm** [comment]
It could be something as simple as how you added the masks in the yml, or it could be that the files were created just fine, but the report didn't catch them.

Do the correct figures exist in the images folder?

**Comment by DrYool** [comment]
Hi Lee,

Erm, the figures for the missing masks get generated OK, but they have no data on them. You can see what I mean here ...

   https://gws-access.jasmin.ac.uk/public/esmeval/CompareReports/bgcval2/ayool/ukesm_bgcval_alkalinity/#OtherPlots-Iron

I half-remember there being some errors in the output report about the missing masks, but I'm not on JASMIN at the moment so can't check.

It could be that I named the masks incorrectly, but I followed what I took to be the convention - and I've certainly got some masks as I'd expect.

Andrew.


-----

Dr. Andrew Yool | MEDUSA lead developer
Room 256/19, National Oceanography Centre, European Way, Southampton, SO14 3ZH

t: +44 (0)23 8059 6339 | e: ***@***.***
noc.ac.uk<https://noc.ac.uk/> | Follow<https://linktr.ee/nocnews> | Support<https://noc.ac.uk/about-us/giving/donate> | Subscribe<https://noc.ac.uk/about-us/nocmail>

[https://noc.ac.uk/files/LOGO-2020---NOC-email-logo.png]<https://noc.ac.uk/>

My normal office hours are Tuesday-Friday | Monday is my non-working day
________________________________

**Comment by ledm** [comment]
Interesting! Would you be able to make a bgcval2 branch and share this code to github so I can help debug it?

**Comment by DrYool** [comment]
Hi Lee,

Not at this precise moment, but I could try to have a look tomorrow. I'm not confident that I can make and share a branch, but I can almost certainly dig up the relevant files that I modified. As I recall, all I did was add the masks to the request.

Would that be OK?

Andrew.


-----

Dr. Andrew Yool | MEDUSA lead developer
Room 256/19, National Oceanography Centre, European Way, Southampton, SO14 3ZH

t: +44 (0)23 8059 6339 | e: ***@***.***
noc.ac.uk<https://noc.ac.uk/> | Follow<https://linktr.ee/nocnews> | Support<https://noc.ac.uk/about-us/giving/donate> | Subscribe<https://noc.ac.uk/about-us/nocmail>

[https://noc.ac.uk/files/LOGO-2020---NOC-email-logo.png]<https://noc.ac.uk/>

My normal office hours are Tuesday-Friday | Monday is my non-working day
________________________________

**Comment by ledm** [comment]
Good plan! Happy to go over the process in a teams meeting so we can screen share, perhaps?
